### PR TITLE
RestoreDashboards: Populate page

### LINKED ISSUE #88389: RestoreDashboards: List deleted dashboards
- Trash page should list the dashboards that have been deleted
 - Deleted dashboards are returned by the Search API with `?deleted=true`, so the UI should reuse the SearchView and (Trash)SearchStateManager

### PR DESCRIPTION
**Pull request by L-M-K-B**
**What is this feature?**
After creating a dedicated section for it I adjust the state management of search view in order to get a list of deleted dashboards

Fixes #88389

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/trash-section/TrashPage.tsx [hunk 1]
```diff
@@ -1,15 +1,85 @@
...
+import { useDispatch } from '../../types';
+import { Props as DashboardRouteComponentsProps } from '../browse-dashboards/BrowseDashboardsPage';
+import { BrowseView } from '../browse-dashboards/components/BrowseView';
+import { SearchView } from '../browse-dashboards/components/SearchView';
+import { getFolderPermissions } from '../browse-dashboards/permissions';
+import { setAllSelection } from '../browse-dashboards/state';
+
+import { useTrashStateManager } from './useTrashStateManager';
+
+const TrashPage = memo(({ match }: DashboardRouteComponentsProps) => {
```
Comment (eledobleefe) [inline public/app/features/trash-section/TrashPage.tsx:19]: You need to update this, right?

File: public/app/features/trash-section/TrashPage.tsx [hunk 2]
```diff
@@ -1,15 +1,85 @@
...
+            isSearching ? (
+              <SearchView
+                canSelect={canSelect}
+                width={width}
+                height={height}
+                searchStateManager={stateManager}
+                searchState={searchState}
+              />
+            ) : (
+              <BrowseView canSelect={canSelect} width={width} height={height} folderUID={folderUID} />
```
Comment (joshhunt) [inline public/app/features/trash-section/TrashPage.tsx:75]: This page is always 'Searching' (as in, using the Search API), so we should always render SearchView.

**Review by joshhunt — COMMENTED** [review]
Just need to remove BrowseView from TrashPage, and then I think this is mergable!

File: public/app/features/trash-section/TrashPage.tsx (hunk 1 above)
Comment (L-M-K-B) [reply public/app/features/trash-section/TrashPage.tsx:19]: Yes, I just merged the renaming PR and now there are conflicts in this one.

**Review by joshhunt — APPROVED** [review]
Nice stuff!

**Pull request merged**